### PR TITLE
Add dataset for monitoring schema errors over time

### PR DESCRIPTION
**Pull request by acmiyaguchi**
This creates a table called `monitoring.schema_error_counts_v1`, which aggregates hourly schema validation errors broken down by the path of the validation error and the name of the decoder that processed the data.

The `error_rank` column allows for efficient filtering, e.g.

```sql
select *
from monitoring.schema_error_counts_v1
where document_type = 'main' and error_rank <= 5
```

Row | document_namespace | document_type | hour | job_name | path | error_count | error_rank |  
-- | -- | -- | -- | -- | -- | -- | -- | --
1 | telemetry | main | 2019-09-25 00:00:00 UTC | decoder-0-0924125001-29254cb | #/environment/build/buildId | 766 | 1 |  
2 | telemetry | main | 2019-09-25 01:00:00 UTC | decoder-0-0924125001-29254cb | #/environment/build/buildId | 704 | 1 |  
3 | telemetry | main | 2019-09-25 02:00:00 UTC | decoder-0-0924125001-29254cb | #/environment/build/buildId | 2062 | 1 |  

This can be broken down by both `job_name` and `path` to create the following dashboard.

image
[Source](https://datastudio.google.com/reporting/11XyjATk_wv1kX-gMLk3bhTDgdNkj4I9D/page/Lj23)

### CODE REVIEW

**Comment by acmiyaguchi** [comment]
Running into an issue with the star operator on the `payload_bytes_error` dataset.

```
 [{'code': 403, 'errors': [{'message': 'Access Denied: BigQuery BigQuery: Permission denied for table: moz-fx-data-shared-prod:payload_bytes_error.*', 'domain': 'global', 'reason': 'accessDenied'}], 'response': {'headers': {'alt-svc': 'quic=":443"; ma=2592000; v="46,43",h3-Q048=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000', 'cache-control': 'private', 'content-encoding': 'gzip', 'content-type': 'application/json; charset=UTF-8', 'date': 'Wed, 23 Oct 2019 20:00:22 GMT', 'server': 'ESF', 'transfer-encoding': 'chunked', 'vary': 'Origin, X-Origin, Referer', 'x-content-type-options': 'nosniff', 'x-frame-options': 'SAMEORIGIN', 'x-xss-protection': '0'}}, 'message': 'Access Denied: BigQuery BigQuery: Permission denied for table: moz-fx-data-shared-prod:payload_bytes_error.*'}]
```

**Comment by whd** [comment]
This fails with a permissions error because it is contingent on https://github.com/mozilla/bigquery-etl/issues/360.

File: templates/monitoring/schema_error_counts_v1/view.sql [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    SPLIT(error_message, ":")[OFFSET (1)] AS path,
+    COUNT(*) AS error_count,
+    ROW_NUMBER() OVER (
+      PARTITION BY
+        hour,
+        document_namespace,
+        document_type
+      ORDER BY
+        COUNT(*) DESC
+    ) AS error_rank
```
Comment (relud) [inline templates/monitoring/schema_error_counts_v1/view.sql]: I could have sworn you aren't allowed to use window functions in conjunction with a group by statement. please confirm this actually works before merging.

File: templates/monitoring/schema_error_counts_v1/view.sql [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  FROM
+    extracted
+  GROUP BY
+    1,
+    2,
+    3,
+    4,
+    5
```
Comment (relud) [inline templates/monitoring/schema_error_counts_v1/view.sql]: please name the columns here, for readability

File: templates/monitoring/schema_error_counts_v1/view.sql (hunk 1 above)
Comment (acmiyaguchi) [reply templates/monitoring/schema_error_counts_v1/view.sql]: Yep, I've been building this in the BigQuery console, so this should work.

**Pull request merged**